DigitalLearningSolutions.Web/Controllers/LoginController.cs

```diff
@@ -67,14 +67,15 @@ public async Task<IActionResult> Index(LoginViewModel model)
                 return View("Index", model);
             }
 
-            if (adminUser != null && adminUser.IsLocked)
+            var (verifiedAdminUser, verifiedDelegateUsers) =
+                loginService.VerifyUsers(model.Password!, adminUser, delegateUsers);
+
+            if (adminUser != null && adminUser.IsLocked && verifiedDelegateUsers.Count == 0)

```

Comment:
I'm not 100% sure, but I think this allows the user to effectively bypass the lock in the following example:

1. A user has an admin account and at least one delegate account. These accounts share the same password.
2. Only the admin account becomes locked from trying to log in to that account 5 times with the wrong password, while the delegate account(s) remain untouched. (This might get counteracted if the Delegate Account locking is implemented, but I think even then users can specifically log in to a single admin account, even if it has the same centre/password as a delegate account, by using the admin-specific Login column)
3. They then try to log in using the email address tied to all accounts, and they use the correct password.
4. When it reaches this if statement, verifiedDelegateUsers.Count > 0, so this is skipped.
5. Seeing as how the password matched their Admin account password, verifiedAdminUser is not null here.
6. Assuming the rest of their account details are suitable for logging in, I think they will be logged in as with their Delegate + Admin account (or at least it will be an option on the Choose a Centre Screen).

Does that example hold water? Or am I missing something here? If this is the case, the unit test will probably need changing as well.

Reply:
Yes I think that might be true, but it might be worth having a quick chat about it in the morning to make sure we get the correct solution.

I think it might be as simple as 
```
if (adminUser != null && adminUser.IsLocked && verifiedDelegateUsers.Count != 0) 
{
    verifiedAdminUser = null;
}
```

Reply:
Happy to have a chat whenever you are ready. I think that solution would work fine, don't know if its worth nesting it inside the above commented line and making it something like
```
if (adminUser != null && adminUser.IsLocked) 
{
    if (verifiedDelegateUsers.Count != 0) 
    {
        verifiedAdminUser = null;
    } else {
        // the other code for redirecting to failed login page
    }
}
```